app/shared/actions/blockchains/wax/claimgbmrewards.js

```diff
@@ -0,0 +1,95 @@
+import * as types from '../../types';
+import * as AccountActions from '../../accounts';
+import eos from '../../helpers/eos';
+
+export function claimgbmrewards() {
+  return async (dispatch: () => void, getState) => {
+    const {
+      connection,
+      settings
+    } = getState();
+
+    const currentAccount = settings.account;
+
+    dispatch({
+      payload: { connection },
+      type: types.SYSTEM_WAX_CLAIMGBMREWARDS_PENDING
+    });
+
+    const { account } = settings;
+    const [, authorization] = connection.authorization.split('@');
+
+    const eosobj = eos(connection, true, true);
+    let method = 'transact';
+    let params = {
+      blocksBehind: 3,
+      expireSeconds: 30,
+    };
+    let contract;
+    if (!eosobj[method]) {
+      contract = await eosobj.getAbi('eosio');
+      if (contract
+        && contract.account_name
+        && contract.abi
+      ) {
+        eosobj.fc.abiCache.abi(contract.account_name, contract.abi);
+      }
+      method = 'transaction';
+      params = {
+        broadcast: connection.broadcast,
+        expireInSeconds: connection.expireInSeconds,
+        sign: connection.sign
+      };
+    }
+
+    return eosobj[method]({
+      actions: [
+        {
+          account: 'eosio',
+          name: 'claimgbmvote',
+          authorization: [{
+            actor: account,
+            permission: authorization
+          }],
+          data: {
+            owner: currentAccount
+          }
+        },
+        {
+          account: 'eosio',
+          name: 'claimgenesis',
+          authorization: [{
+            actor: account,
+            permission: authorization
+          }],
+          data: {
+            claimer: currentAccount
+          }
+        },
+      ]
+    }, params).then((tx) => {
+      setTimeout(() => {
+        dispatch(AccountActions.getCurrencyBalance(currentAccount));
```

Comment:
Does this also chain and update staked EOS values? If not, we might also want to issue the call to do that.

Reply:
Hmm why do we want to update the staked amounts?

Reply:
Because when you claim, you also get tokens staked directly to your account as well as a liquid reward

Reply:
Cool, I didn't know that 😄 I am now calling the getAccount action which will fetch both stake amounts and balances.